### ISSUE TITLE
TOO_MANY_REGISTRATIONS

### ISSUE THREAD

**Issue by javieritis** [OneSignal/OneSignal-Android-SDK#499]
OneSignal android it returns the next error un log:
```java
E/OneSignal: Error Getting Google Registration ID
    java.io.IOException: TOO_MANY_REGISTRATIONS
        at com.google.android.gms.gcm.GoogleCloudMessaging.zza(Unknown Source:92)
        at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source:7)
        at com.onesignal.PushRegistratorGPS$2.run(PushRegistratorGPS.java:167)
        at java.lang.Thread.run(Thread.java:764)
```
and dont receive any notification push...

Can anybody help me

**Comment by jkasten2** [comment]
@javieritis This can happen on a test device or emulator if have a large number of apps that are being reinstall continuously. It shouldn't happen a real device under normal use. See the following for a bit more details. https://stackoverflow.com/a/48439819

**Comment by javieritis** [comment]
@jkasten2 dont use emulator, i use the reals phones nexus 5, nexus 6, and other...

**Comment by chrismarrin** [comment]
```
11-12 16:21:14.907: E/OneSignal(24540): Error Getting FCM Token
11-12 16:21:14.907: E/OneSignal(24540): java.io.IOException: TOO_MANY_REGISTRATIONS
11-12 16:21:14.907: E/OneSignal(24540): 	at com.google.firebase.iid.zzs.zza(Unknown Source:71)
11-12 16:21:14.907: E/OneSignal(24540): 	at com.google.firebase.iid.zzs.zza(Unknown Source:84)
11-12 16:21:14.907: E/OneSignal(24540): 	at com.google.firebase.iid.zzt.then(Unknown Source:4)
11-12 16:21:14.907: E/OneSignal(24540): 	at com.google.android.gms.tasks.zzd.run(Unknown Source:5)
11-12 16:21:14.907: E/OneSignal(24540): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
11-12 16:21:14.907: E/OneSignal(24540): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
11-12 16:21:14.907: E/OneSignal(24540): 	at java.lang.Thread.run(Thread.java:764)
```

I just wanted to mention this for any developers seeing `Google Play Services Library Errors` in your audiences, double check to see if those devices are the ones being used for the Google Play Pre-Launch Report.   I was seeing a group of devices with this error ( and as well on my device, but that was another issue caused by deleting a user in the dashboard ), and was concerned.  Looking into it, each of the device logs from the pre-launch report had this error.  I haven't seen this error on any real device.

So basically, if you're seeing unexplained errors, it could be the Google Pre Launch Report.
